### PR TITLE
Dmulder/key auth value

### PR DESCRIPTION
**Pull request by Firstyear**
Refers #32 - Fix PIN Value support for identity keys. This works on the TPM with it's native lockout features, and soft with argon2id derivation hmaced (peppered) to the machine key.

## Checklist

- [ x ] This pr contains no AI generated code
- [ x ] cargo fmt has been run
- [ x ] cargo clippy has been run and there's no issues
- [ x ] cargo test has been run and passes

### CODE REVIEW

**Comment by dmulder** [comment]
Odd, I see these:
```
2024-03-27T14:05:59.108386Z ERROR kanidm_hsm_crypto: elapsed=Ok(331.406153ms)
2024-03-27T14:05:59.124725Z ERROR kanidm_hsm_crypto: elapsed=Ok(332.024759ms)
2024-03-27T14:05:59.438566Z ERROR kanidm_hsm_crypto: elapsed=Ok(330.107573ms)
2024-03-27T14:05:59.464243Z ERROR kanidm_hsm_crypto: elapsed=Ok(339.426009ms)
```
All the tests pass though. Maybe these should be going to debug, not error?

File: src/lib.rs [hunk 1]
```diff
@@ -38,6 +38,79 @@ pub mod tpm;
...
+        let now = std::time::SystemTime::now();
+
+        argon
+            .hash_password_into(self.value.as_ref(), salt, auth_key.as_mut())
+            .map_err(|argon_err| {
+                error!(?argon_err);
+                TpmError::AuthValueDerivation
+            })?;
+
+        error!(elapsed = ?now.elapsed());
```
Comment (dmulder) [inline src/lib.rs:108]: Looks like it's coming from here. Perhaps this is just a testing statement you forgot to remove?

**Comment by dmulder** [comment]
Otherwise, LGTM. Thanks for doing this! I built this against some test code and it worked as expected.

**Review by dmulder — APPROVED** [review]
LGTM

**Comment by dmulder** [comment]
When you merge, can you issue a release?

File: src/lib.rs (hunk 1 above)
Comment (Firstyear) [reply src/lib.rs:108]: Yeah, it is 🤣

**Comment by Firstyear** [comment]
@dmulder released 0.2.0

**Comment by dmulder** [comment]
> @dmulder released 0.2.0

Sigh, I always forget about the dependency loop on compact-jwt. That also needs to be updated to point to the new hsm-crypto (else none of my code will build).

**Comment by Firstyear** [comment]
If all goes well, I will push my changes to compact-jwt today so we can release them.

**Pull request merged**